commons/src/main/java/io/aiven/kafka/connect/common/source/input/JsonTransformer.java

```diff
@@ -105,21 +120,18 @@ public boolean tryAdvance(final java.util.function.Consumer<? super Object> acti
                 while (line != null) {
                     line = line.trim();
                     if (!line.isEmpty()) {
-                        try {
-                            final JsonNode jsonNode = objectMapper.readTree(line); // Parse the JSON
-                            // line
-                            action.accept(jsonNode); // Provide the parsed JSON node to the stream
-                        } catch (IOException e) {
-                            LOGGER.error("Error parsing JSON record: {}", e.getMessage(), e);
-                        }
+                        // check if it's a valid json
+                        objectMapper.readTree(line);
```

Comment:
Is this parsing the whole file twice? Seems wasteful.

Reply:
this is parsing only the json line, and it would be done only once.

Reply:
One parse is objectMapper.readTree, the other is jsonConverter#toConnectData.

Reply:
Nice one, removed. Updated a unit test for the same.